components/proxy/src/main/java/com/hotels/styx/startup/StyxServerComponents.java

```diff
@@ -107,8 +108,8 @@ private StyxServerComponents(Builder builder) {
                 .map(StyxServerComponents::readHttpHandlers)
                 .orElse(ImmutableMap.of())
                 .forEach((name, record) -> {
-                    RoutingObject handler = routingObjectFactory.build(ImmutableList.of(name), record);
-                    routeObjectStore.insert(name, new RoutingObjectRecord(record.type(), record.config(), handler))
+                    RoutingObjectAdapter adapter = new RoutingObjectAdapter(routingObjectFactory.build(ImmutableList.of(name), record));
```

Comment:
is adapter a good name here?

Reply:
It adapts a `RoutingObject` instance to Styx core, adding some functionality to it (load balancing metric).

I tried to google for some industry standard names for this use case but drew a blank. I'm open for any sensible proposals.

Reply:
So I would say that in default OO patterns this is not an adapter but a Decorator.  An Adapter is supposed to bridge two interfaces , whereas Decorators wrap an existing interface providing additional functionality.

Reply:
Okay. Decorator it is!